New rustc nightly with llvm-10 breaks script_plugins

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.0.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.1.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.10.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.11.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.12.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.13.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.14.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.15.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.2.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.3.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.4.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.5.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.6.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.7.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.8.rcgu.o" "/repo/target/debug/deps/script_plugins.script_plugins.blvvqtbe-cgu.9.rcgu.o" "-o" "/repo/target/debug/deps/libscript_plugins.so" "-Wl,--version-script=/tmp/rustczV5bvX/list" "/repo/target/debug/deps/script_plugins.a9q7pget1xzpz9k.rcgu.o" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/repo/target/debug/deps" "-L" "/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-L" "/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-lrustc_driver-fefef43299f39010" "-Wl,--start-group" "-L" "/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-lstd-265ff16c7b6b86f9" "-Wl,--end-group" "-Wl,-Bstatic" "/tmp/rustczV5bvX/libcompiler_builtins-a1755b3e88842f5d.rlib" "-Wl,-Bdynamic" "-lLLVM-10-rust-1.45.0-nightly" "-lutil" "-ldl" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-ldl" "-lutil" "-shared" "-fuse-ld=gold"
  = note: /usr/bin/ld.gold: error: cannot find -lLLVM-10-rust-1.45.0-nightly
          collect2: error: ld returned 1 exit status
          
error: aborting due to previous error
error: could not compile `script_plugins`.
```
This might be something we need to report upstream.

--- a/components/derive_common/cg.rs
+++ b/components/derive_common/cg.rs
@@ -7,7 +7,7 @@ use proc_macro2::{Span, TokenStream};
 use quote::TokenStreamExt;
 use syn::{self, AngleBracketedGenericArguments, Binding, DeriveInput, Field};
 use syn::{GenericArgument, GenericParam, Ident, Path};
-use syn::{PathArguments, PathSegment, QSelf, Type, TypeArray};
+use syn::{PathArguments, PathSegment, QSelf, Type, TypeArray, TypeGroup};
 use syn::{TypeParam, TypeParen, TypePath, TypeSlice, TypeTuple};
 use syn::{Variant, WherePredicate};
 use synstructure::{self, BindStyle, BindingInfo, VariantAst, VariantInfo};
@@ -184,6 +184,10 @@ where
             path: map_type_params_in_path(path, params, f),
         }),
         Type::Paren(ref inner) => Type::from(TypeParen {
+            elem: Box::new(map_type_params(&inner.elem, params, f)),
+            ..inner.clone()
+        }),
+        Type::Group(ref inner) => Type::from(TypeGroup {
             elem: Box::new(map_type_params(&inner.elem, params, f)),
             ..inner.clone()
         }),

--- a/python/servo/command_base.py
+++ b/python/servo/command_base.py
@@ -986,7 +986,13 @@ install them, let us know by filing a bug!")
             installed = check_output(
                 ["rustup", "component", "list", "--installed", "--toolchain", toolchain]
             )
-            for component in set(rustup_components or []) | {"rustc-dev"}:
+            required_components = {
+                # For components/script_plugins, https://github.com/rust-lang/rust/pull/67469
+                "rustc-dev",
+                # https://github.com/rust-lang/rust/issues/72594#issuecomment-633779564
+                "llvm-tools-preview",
+            }
+            for component in set(rustup_components or []) | required_components:
                 if component.encode("utf-8") not in installed:
                     check_call(["rustup", "component", "add", "--toolchain", toolchain, component])
 

